controller/login.go

```diff
@@ -112,6 +113,63 @@ func (c *LoginController) Authorize(ctx *app.AuthorizeLoginContext) error {
 	return c.auth.Perform(ctx, oauth, brokerEndpoint, entitlementEndpoint, profileEndpoint, whitelist, c.configuration.GetAuthNotApprovedRedirect())
 }
 
+// getEntitlementResourceRequestPayload creates the object which would have the information about which spaces/resources
+// the entitlements' info would need to be fetched for.
+
+func (c *LoginController) getEntitlementResourceRequestPayload(ctx context.Context, token *string) (*auth.EntitlementResource, error) {
+	loggedInIdentityID, err := c.tokenManager.Extract(*token)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "unable to get ID from access token")
+		return nil, errors.NewInternalError(ctx, errs.Wrap(err, "unable to get ID from access token"))
+	}
+
+	// get the user object as well for this identity
+	queryResult, err := c.identityRepository.Query(account.IdentityFilterByID(loggedInIdentityID.ID), account.IdentityWithUser())
+	if err != nil || len(queryResult) == 0 {
+		log.Error(ctx, map[string]interface{}{
+			"err":         err,
+			"identity_id": *loggedInIdentityID,
+		}, "unable to query Identity")
+		return nil, errors.NewInternalError(ctx, errs.Wrap(err, "unable to query Identity"))
+	}
+	loggedInIdentity := queryResult[0]
+	contextInfoLoggedInIdentity := loggedInIdentity.User.ContextInformation
+	_, recentSpacesPresent := contextInfoLoggedInIdentity["recentSpaces"]
+	if contextInfoLoggedInIdentity == nil || !recentSpacesPresent {
+		log.Warn(ctx, map[string]interface{}{
+			"identity_id": *loggedInIdentityID,
+		}, "unable to find recentSpaces in ContextInformation")
+		return nil, nil
+	}
+
+	var spacesToGetEntitlementsFor []auth.ResourceSet
+	for _, v := range contextInfoLoggedInIdentity["recentSpaces"].([]interface{}) {
+		recentSpaceID, ok := v.(string)
+		if !ok {
+			log.Warn(ctx, map[string]interface{}{
+				"identity_id": *loggedInIdentityID,
+			}, "unable to find a string uuid in recentSpaces in contextInformation")
+			return nil, nil
+		}
+		spacesToGetEntitlementsFor = append(spacesToGetEntitlementsFor, auth.ResourceSet{Name: recentSpaceID}) // pass by reference?
```

Comment:
That's in another PR : https://github.com/fabric8-services/fabric8-wit/pull/1458, will take in those changes after it's merged. ( will merge it after keycloak prod is upgraded ).